data/src/test/java/com/linkedin/data/schema/TestPdlBuilder.java

```diff
@@ -0,0 +1,41 @@
+package com.linkedin.data.schema;
+
+import com.linkedin.data.DataMap;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.util.Collections;
+import java.util.Map;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+
+public class TestPdlBuilder {
+
+  @DataProvider
+  private static Object[][] propertiesMapProvider()
+  {
+    DataMap properties1 = new DataMap();
+    properties1.put("empty", new DataMap());
+    DataMap properties2 = new DataMap();
+    properties2.put("validate", properties1);
+    DataMap properties3 = new DataMap();
+    DataMap nestedMap = new DataMap(properties2);
+    nestedMap.putAll(properties1);
+    properties3.put("nested", nestedMap);
+    return new Object[][]{{properties1, "@empty = {}\n"},
+        {properties2, "@validate.empty = {}\n"}
+        //TODO Add test case for multiple properties in a map level once iteration logic is fixed to be deterministic
```

Comment:
It's fine that each test (for now) has one root-level property, but none of them branch. Thus, these cases don't actually test your changes. You'll need to make this a little more comprehensive.

Reply:
as discussed this is not possible right now, as ordering is not deterministic.